test/sslapitest.c

```diff
@@ -3325,7 +3334,7 @@ static int find_session_cb(SSL *ssl, const unsigned char *identity,
     if (find_session_cb_cnt > 2)
         return 0;
 
-    if (serverpsk == NULL)
+    if (serverpsk == NULL || !SSL_SESSION_up_ref(serverpsk))
```

Comment:
Leaks `serverpsk` in the event that the `if` block below evaluates to true and we return 1 early.